Automattic/wp-calypso
# Components: Port and remove unused SiteUsersFetcher

#### Changes proposed in this Pull Request

* Concierge: Use react-query instead of SiteUsersFetcher
* Remove unused `SiteUsersFetcher`.

This depends on #51882 and has to be rebased (with https://github.com/Automattic/wp-calypso/commit/66a9a5a9fb6f4f804c0f2a3956efed75cb09dfea removed) after it lands.

Related to #24150.

#### Testing instructions

* Run the branch locally.
* In `is-rebrand-cities-site.js` edit `REBRAND_CITIES_ACCOUNT_USERNAME` to your username 😉 
* Go to `/me/concierge`
* Pick a simple WP.com site. 
* Buy a concierge session if you don't have one.
* After finishing the purchase, navigate to `/me/concierge/:site/book`
* Explore the components, and verify that after loading, the `signup` prop in the `InfoStep` component has a prop `isRebrandCitiesSite` that is equal to `true`.
* Verify there are no errors introduced and users loading works correctly. 
* Verify `SiteUsersFetcher` isn't used anywhere anymore.
